src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -24,20 +25,22 @@ var {
   ClassComponent,
   HostRoot,
 } = require('ReactTypeOfWork');
+const {
+  createCursor,
+  pop,
+  push,
+  reset,
+} = require('ReactFiberStack');
 
 if (__DEV__) {
   var checkReactTypeSpec = require('checkReactTypeSpec');
 }
 
-let index = -1;
-const contextStack : Array<Object> = [];
-const didPerformWorkStack : Array<boolean> = [];
+let contextStackCursor : StackCursor<?Object> = createCursor((null: ?Object));
+let didPerformWorkStackCursor : StackCursor<?boolean> = createCursor((null: ?boolean));
```

Comment:
I thought about doing that, but then I'd lose the ability to differentiate between "empty" and default-state. Which isn't really important in this case but... slightly more compelling is that it would also mean the cursor would have asymmetric behavior- `false` by default, and then `null` after the stack gets reset (since we don't track and fall back to the default value). It seemed worth doing the cast to avoid that, but I don't feel strongly about it.